version-compatibility/forkless-upgrade/src/backward_compatibility.rs

```diff
@@ -109,7 +112,76 @@ async fn latest_binary_is_backward_compatible_and_follows_blocks_created_by_gene
 
     // Then
     for i in 0..BLOCKS_TO_PRODUCE {
-        let _ = tokio::time::timeout(Duration::from_secs(120), imported_blocks.next())
+        let _ = tokio::time::timeout(BLOCK_INCLUSION_TIMEOUT, imported_blocks.next())
+            .await
+            .expect(format!("Timed out waiting for block import {i}").as_str())
+            .expect(format!("Failed to import block {i}").as_str());
+    }
+}
+
+#[tokio::test]
+async fn latest_binary_is_backward_compatible_and_follows_blocks_created_by_v36_binary() {
+    // Given
+    let v36_keypair = SecpKeypair::generate();
+    let hexed_secret = hex::encode(v36_keypair.secret().to_bytes());
+    let v36_port = "30333";
+    let v36_node = Version36FuelCoreDriver::spawn(&[
+        "--service-name",
+        "V36Producer",
+        "--debug",
+        "--poa-instant",
+        "true",
+        "--consensus-key",
+        POA_SECRET_KEY,
+        "--snapshot",
+        IGNITION_TESTNET_SNAPSHOT,
```

Comment:
This snapshot using the state transition function from `fuel-core 0.26.0`. Maybe we want to add a new snapshot for `fuel-core 0.36`?=)